test/functional/test_framework/test_node.py

```diff
@@ -229,6 +230,23 @@ def is_node_stopped(self):
     def wait_until_stopped(self, timeout=BITCOIND_PROC_WAIT_TIMEOUT):
         wait_until(self.is_node_stopped, timeout=timeout)
 
+    @contextlib.contextmanager
+    def assert_debug_log(self, expected_msgs):
```

Comment:
I was surprised to see that this code actually works without adding RPCs before and after the yield to flush the log. But apparently there is a `setbuf` call in the logging code which disables buffering. Might be worth noting in a comment.